mmengine/fileio/backends/oss_backend.py

```diff
@@ -0,0 +1,809 @@
+        try:
+            import oss2
+            self.oss2 = oss2
+        except ImportError:
+            raise ImportError('Please install OSS_client to enable '
+                              'OSSBackend.')
+        self._client = self.oss2.Auth(
+            access_key_id=access_key_id, access_key_secret=access_key_secret)
+
+        assert isinstance(path_mapping, dict) or path_mapping is None
+        self.path_mapping = path_mapping
+        # Use to parse bucket and obj_name
+        self.parse_bucket = re.compile('oss://(.+?)/(.+?)/(.*)')
+
+    def _map_path(self, filepath: Union[str, Path]) -> str:
+        """Map ``filepath`` to a string path whose prefix will be replaced by
+        :attr:`self.path_mapping`.
+
+        Args:
+            filepath (str or Path): Path to be mapped.
+        """
+        filepath = str(filepath)
+        if self.path_mapping is not None:
+            for k, v in self.path_mapping.items():
+                filepath = filepath.replace(k, v, 1)
+        return filepath
+
+    def _format_path(self, filepath: str) -> str:
+        """Convert a ``filepath`` to standard format of OSS oss.
+
+        If the ``filepath`` is concatenated by ``os.path.join``, in a Windows
+        environment, the ``filepath`` will be the format of
+        'oss://bucket_name\\image.jpg'. By invoking :meth:`_format_path`, the
+        above ``filepath`` will be converted to 'oss://bucket_name/image.jpg'.
+
+        Args:
+            filepath (str): Path to be formatted.
+        """
+        return re.sub(r'\\+', '/', filepath)
+
+    def _parse_path(self, filepath: Union[str, Path]) -> Tuple[str, str, str]:
+        """Parse endpoint, bucket, and object name from a given ``filepath``.
+
+        Args:
+            filepath (str or Path): Path to read data.
+
+        Returns:
+            tuple[str, str, str]: The first item is the endpoint meaning
+            access domain or CNAME. The second is the bucket name
+            of oss. The last is obj_name which is the object's relative
+            path to the bucket.
+        """
+        filepath = self._map_path(filepath)
+        filepath = self._format_path(filepath)
+        parse_res = self.parse_bucket.findall(filepath)
+        if not parse_res:
+            raise ValueError(
+                f"The input path '{filepath}' format is incorrect. "
+                'An example: oss://oss-cn-hangzhou.aliyuncs.com/mmengine/.*')
+        endpoint, bucket, obj_name = parse_res[0]
+        return endpoint, bucket, obj_name
+
+    def _bucket_instance(self, endpoint: str, bucket_name: str):
+        """get bucket instance from a given ``endpoint`` and ``bucket_name``.
+
+        Args:
+            endpoint (str): aliyun Endpoint.
+            bucket_name (str): bucket name of OSS
+
+        Returns:
+            Bucket: An instance of OSS.
+        """
+
+        bucket = self.oss2.Bucket(self._client, endpoint, bucket_name)
```

Comment:
```suggestion
        bucket = self.oss2.Bucket(self._auth, endpoint, bucket_name)
```

Reply:
Will `self.oss2.bucket` return the same object if `self._auth`, `endpoint`, and `bucket_name` are the same as before?

Reply:
yes, Do I need to modify all `self._client` to `self._auth`.

Reply:
Yes